File: 진우/11weeks/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.cc
#include <iostream>
#include <string>
#include <set>
using namespace std;

int main() {
    string S;
    cin >> S;

    int len = S.length();
    set<string> arr; // 서로 다른 부분 문자열 저장

    // i = 시작 위치
    for (int i = 0; i < len; i++) {
        string word = ""; // i부터 j까지 이어붙여 만들 부분 문자열

        // j = 현재 문자
        for (int j = i; j < len; j++) {
            word += S[j];    // 한 글자씩 이어서 부분 문자열 생성
            arr.insert(word); // set에 추가 (중복 자동 제거)
        }
    }

    cout << arr.size() << "\n";
    return 0;
}

File: 진우/11weeks/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.py
S = input().strip()
n = len(S)
arr = set()  # 서로 다른 부분 문자열 저장

for i in range(n):       # 시작 위치
    word = ""           # 부분 문자열 누적 변수
    for j in range(i, n):   # 현재 끝 위치
        word += S[j]        # 한 글자씩 이어붙이기
        arr.add(word)       # set에 추가 (중복 자동 제거)

print(len(arr))
